go/apps/ctrl/services/build/backend/depot/create_build.go

```diff
@@ -0,0 +1,277 @@
+package depot
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"buf.build/gen/go/depot/api/connectrpc/go/depot/core/v1/corev1connect"
+	"connectrpc.com/connect"
+	"github.com/depot/depot-go/build"
+	"github.com/depot/depot-go/machine"
+	"github.com/docker/cli/cli/config/configfile"
+	"github.com/docker/cli/cli/config/types"
+	"github.com/moby/buildkit/client"
+	"github.com/moby/buildkit/session"
+	"github.com/moby/buildkit/session/auth/authprovider"
+
+	corev1 "buf.build/gen/go/depot/api/protocolbuffers/go/depot/core/v1"
+	cliv1 "github.com/depot/depot-go/proto/depot/cli/v1"
+	ctrlv1 "github.com/unkeyed/unkey/go/gen/proto/ctrl/v1"
+	"github.com/unkeyed/unkey/go/pkg/assert"
+	"github.com/unkeyed/unkey/go/pkg/db"
+)
+
+// CreateBuild orchestrates the container image build process using Depot.
+//
+// Steps:
+//
+//	Get or create Depot project
+//	Register a new build with Depot
+//	Acquire a build machine
+//	Connect to the buildkit instance
+//	Prepare build context and configuration
+//	Execute the build with status logging
+//	Return build metadata
+func (s *Depot) CreateBuild(
+	ctx context.Context,
+	req *connect.Request[ctrlv1.CreateBuildRequest],
+) (*connect.Response[ctrlv1.CreateBuildResponse], error) {
+	if err := assert.All(
+		assert.NotEmpty(req.Msg.BuildContextPath, "build_context_path is required"),
+		assert.NotEmpty(req.Msg.UnkeyProjectId, "unkey_project_id is required"),
+	); err != nil {
+		return nil, connect.NewError(connect.CodeInvalidArgument, err)
+	}
+
+	buildPlatform := strings.TrimPrefix(s.buildPlatform, "/")
+	parts := strings.Split(buildPlatform, "/")
+
+	if err := assert.All(
+		assert.Equal(len(parts), 2, fmt.Sprintf("invalid build platform format: %s (expected format: linux/amd64)", s.buildPlatform)),
+		assert.Equal(parts[0], "linux", fmt.Sprintf("unsupported OS: %s (only linux is supported)", parts[0])),
+	); err != nil {
+		return nil, connect.NewError(connect.CodeInvalidArgument, err)
+	}
+
+	platform := buildPlatform
+	architecture := parts[1]
```

Comment:
this comes from initializing the control plane right? 

We should validate it on startup then and split it up into arch and platform, and not re-validate it each time we create a deployment.